compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -101,6 +102,18 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
             }
         };
 
+        if let hir::Attribute::Parsed(p) = attr {
+            match p {
+                AttributeKind::Repr(reprs) => {
+                    codegen_fn_attrs.alignment = reprs
+                        .iter()
+                        .find_map(|(r, _)| if let ReprAlign(x) = r { Some(*x) } else { None });
+                }
+
+                _ => {}
```

Comment:
I'm torn. Most attrs are not relevant here, but it would also be convenient to, when adding a new attr, immediately have the compiler tell you to handle it here or add a new variant.

unsure if there's some other way than exhaustively matching. We could in the future require marking which attrs are codegen attrs (or even group them in a variant with sub-variants). 

Maybe just make this exhaustive and we'll see how it goes in practice

Reply:
grouping them is certainly a possibility. For a while my enum had a grouping actually, for all `rustc_*` attributes but I didn't convert enough of those in this PR to keep it around. I do like that approach

Reply:
the problem is, what do you return out of this method? that's a more general problem, we don't have variant types. `get_repr()` should return something like `Option<Vec<ReprAttr>>` but other fns have different signatures. This is a more general problem with this entire approach, one that petrochenkov was a little afraid of I think. It's semantically nicer to have an enum, but it's hard to name a variant without its data, or data without its variant. That's why this is a macro, so you can use a pattern.

Reply:
wrong thread?

Reply:
@oli-obk I think this is not for this PR. But we can do this in the future, I'll make an issue for it ok?